packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -106,17 +121,26 @@ function DocSidebarItemCategory({
         'menu__list-item--collapsed': collapsed,
       })}
       key={label}>
-      <a
+      <Link
+        to={initialLink}
         className={clsx('menu__link', {
           'menu__link--sublist': collapsible,
           'menu__link--active': collapsible && isActive,
-          [styles.menuLinkText]: !collapsible,
+          [styles.menuLinkText]: !collapsible && initialLink === '',
         })}
+        {...(isInternalUrl(initialLink)
+          ? {
+              isNavLink: true,
+              onClick: onItemClick,
+            }
+          : {
+              target: '_blank',
```

Comment:
You do not need to manually set attributes for external links, please see this commit https://github.com/facebook/docusaurus/pull/4242/files#diff-4fcb9710514433c26ba9bd767a61ab349d919e22ee776b78762e3d5f9e2e2e9d for more details.

Reply:
Should be addressws in a2e87935